srcts/src/shiny/bind.ts

```diff
@@ -119,13 +119,22 @@ const bindingsRegistry = (() => {
       })
       .join("\n");
 
+    const finalMsg = `The following ${
+      duplicateIds.size === 1 ? "ID was" : "IDs were"
+    } repeated:\n${duplicateIdMsg}`;
+
+    // If we're not in dev mode, just log a warning and continue
+    if (!Shiny.inDevMode()) {
+      console.warn(finalMsg);
```

Comment:
Can we add a hint that this warning is coming from Shiny here? IIRC we use (or used) this prefix for other console-logged messages.

```suggestion
      console.warn("[shiny] " + finalMsg);
```